docs/rfc/identity_proofs.md

```diff
@@ -0,0 +1,214 @@
+      "0x32be343b94f860124dc4fee278fdcbd38c102d88".
+
+* `expiration` (optional)
+    * `created`
+
+      Creation timestamp of the claim, in seconds since 1970-01-01T00:00:00Z.
+
+    * `expires`
+
+      Seconds relative to `created`, after which the claim should no longer be
+      considered.
+
+* `proof` (optional)
+
+  A URL to assist verification tooling in retrieving the proof from the external
+  system. This is mainly a convenience, and obviously requires creation of a new
+  revision after the fact.
+
+## Proof Generation
+
+The above claim payload is committed to the identity history as a new revision.
+Technically, this revision needs to be approved by all key delegations for
+verification to pass later on, but since we assume that eligible keys are held
+by the same person, it may be acceptable to publish the proof right away for
+user experience reasons.
+
+The actual proof consists of the following tuple:
+
+    (root, revision, public key, signature)
+
+> TODO: include a protocol bit (denoting "git"), for forwards compat?
+
+The values `root`, `revision`, and `public key` are specified in
+[identities][ids], and it is RECOMMENDED to follow the serialisation formats
+devised there. `signature` is the Ed25519 signature over `revision`, in much the
+same way as the actual revision is signed. All values can thus be obtained by
+inspecting the identity storage.
+
+It is beyond the scope of this document to devise the exact external format to
+serialise the tuple into, as this is expected to vary from service to service.
+
+## Revocation
+
+A claim can be revoked by creating a new identity revision which simply does not
+contain the claim payload. Likewise, a later claim describing the same `(service
+name, account)` mapping invalidates an earlier one.
+
+> NOTE: claiming multiple accounts per service is permissible in principle, but
+> might be undesirable for certain services. In this case, service-specific
+> post-validation is required.
+
+## Verification
+
+1. Given the 4-tuple specified [above][#proof-generation], it is first verified
+   that the signature is valid for the given `revision` and `public key`.
+
+2. If it is, the identity history needs to be resolved from local storage, or
+   the network.
+
+  2.1 If persistence of verification results is available, the identity document
+      at the given `revision` is retrieved, its `root` compared against the
+      supplied one, and it is verified that the given `public key` is a member
+      of the `delegations` set. If expiration is specified in the claim, it is
+      determined if `expiration` seconds already elapsed since `created`
+      according to the clock, in which case verification fails.
+  2.2 Otherwise, the identity verification procedure is run first, after which
+      the history is walked backwards until the `revision` is found. Then, step
+      2.1 is performed.
+
+3. Lastly, the identity history needs to be examined for revocations.
```

Comment:
Could you explain why this is? As in, say that we validated the tip of the history where a new claim is added. Why is it necessary to examine the earlier history for revocations?

Reply:
Ya I guess this can be collapsed into:

* find tip of refs/namespaces/<root>/refs/remotes/<public key>/rad/id 
* verify
* check claim is included
* if yes, walk backwards to verify revision is an ancestor, and signing key was eligible 

Originally I thought to avoid explicit revocation, but then, it is hard to prevent _accidental_ removal when the identity document is updated, which would result in an undefined state.

Reply:
So if the revision is older than the latest tip, you'd have to look at its children for revocations, right?

Reply:
No actually not, that's where the confusion stems from: if we define that revocation is equivalent to creating a new revision which simply does not contain the "claim" payload, then we only need to look at the tip. If, however, the revision from the proof 4-tuple is not in the ancestry path, we either have a fork or not the latest tip.

Reply:
> if we define that revocation is equivalent to creating a new revision which simply does not contain the "claim" payload

But isn't that how we are [defining revocation](https://github.com/radicle-dev/radicle-link/pull/525/files#diff-5389480c1bf9cf91c20b588fb42a26972437745bc31fa234d6be995afbaa001bR130-R131)? :smile: Or am I getting further confused? :grimacing:

Reply:
Yes, exactly. If doc at revision `A` verifies, but does not contain the claim, the proof can be rejected right away. To _report_ that it is a revocation (ie. it was valid at a previous point in time), revision `B` (from the proof tuple) must be an ancestor of `A` **and** the claim must be present at `A`. Arguably, you can compute this distinction lazily, because the difference between "was never valid" and "was previously valid" is only meaningful to humans.

Reply:
Ok, this is clear. Thanks!